packages/generator/src/renderer.ts

```diff
@@ -52,18 +52,23 @@ interface IBuildConfig {
   shopperAuthApi: string;
   exchangeDeploymentRegex: RegExp;
 }
+
 /**
- * Information used to generate APICLIENTS.md.
+ * Api Document with metadata
+ * TODO: This should be moved into raml-toolkit
  */
-export interface IApiClientsInfo {
-  model: model.domain.WebApi;
-  family: string;
-  config: RestApi;
-}
+export type DocumentWithMetadataT = {
+  document: model.document.Document;
+  metadata: {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    [key: string]: any;
+  };
+};
```

Comment:
```suggestion
type DocumentMetadata = RestApi & {
  apiSpec: string;
  specName: string;
};
export type DocumentWithMetadataT = {
  document: model.document.Document;
  metadata: DocumentMetadata;
};
```

Reply:
I intentionally left metadata to be whatever you wanted.  Did NOT want to specify specific constraints here.

Reply:
```suggestion
type DocumentMetadata = RestApi & {
  apiSpec: string;
  specName: string;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  [key: string]: any;
};
export type DocumentWithMetadataT = {
  document: model.document.Document;
  metadata: DocumentMetadata;
};
```

With this, known properties are typed while still allowing extensibility.

Reply:
Not terrible worried about this here.  This is definitely a nice enhancement once we move stuff over.  But would rather keep this as simple as possible at this point.  apiSpec and apiName aren't freeform metadata in Brian's new PR, they are known aspects of the class.  Basically would rather keep what is here simpler to make the transition easier.